### PR TITLE
fix(ClickHouse): fix 'Direct select is not allowed'

### PR DESCRIPTION
**Pull request by guidoiaquinti**
## Changes
Fix in preparation for the ClickHouse version upgrade. See [upstream setting](https://github.com/ClickHouse/ClickHouse/blob/ec5148d5a5d28cd7bd4c5ec701beb25b3dae7f1d/src/Core/Settings.h#L100
) for more info.

## How did you test this code?

After

```
$ pytest ee/clickhouse/models/test/test_dead_letter_queue.py
[...]
============================================================================================ 2 passed in 27.45s ============================================================================================
```

Before
````
$ pytest ee/clickhouse/models/test/test_dead_letter_queue.py

[...]

E           clickhouse_driver.errors.ServerException: Code: 620.
E           DB::Exception: Direct select is not allowed. To enable use setting stream_like_engine_allow_direct_select. Stack trace:
[...]
========================================================================================= short test summary info ==========================================================================================
FAILED ee/clickhouse/models/test/test_dead_letter_queue.py::TestDeadLetterQueue::test_direct_table_insert - ee.clickhouse.errors.CHQueryErrorUnknown: Code: 620.
FAILED ee/clickhouse/models/test/test_dead_letter_queue.py::TestDeadLetterQueue::test_kafka_insert - ee.clickhouse.errors.CHQueryErrorUnknown: Code: 620.
============================================================================================ 2 failed in 10.56s ============================================================================================

### CODE REVIEW

**Review by yakkomajuri — APPROVED** [review]
great, thanks!

**Review by hazzadous — APPROVED** [review]
Looks good.

Could we also just drop and create?

File: ee/clickhouse/models/test/test_dead_letter_queue.py [hunk 1]
```diff
@@ -43,8 +45,18 @@
...
+    # We can't truncate a table using Kafka engine but reading from it will delete all the rows
+    truncate_kafka_table_query = "SELECT * FROM kafka_events_dead_letter_queue"
+
+    # Note: ClickHouse version >= 21.12 do not allow direct select for Kafka/RabbitMQ/FileLog engine tables.
+    #       We can pass `stream_like_engine_allow_direct_select` to override this behavior.
+    #       Once we drop support for ClickHouse < 21.12 we can remove this if.
+    version_range = SimpleSpec(">=21.12")
+    current_version = get_clickhouse_version()
+    if current_version in version_range:
+        truncate_kafka_table_query += " SETTINGS stream_like_engine_allow_direct_select = 1"
```
Comment (macobo) [inline ee/clickhouse/models/test/test_dead_letter_queue.py:57]: sync_execute has a settings argument, use that whenever possible.

Also I believe we can just always pass this setting irregardless of the version

Comment (guidoiaquinti) [reply ee/clickhouse/models/test/test_dead_letter_queue.py]: > sync_execute has a settings argument, use that whenever possible.

TIL

> Also I believe we can just always pass this setting irregardless of the version

The query will fail for versions < 21.12 as the setting was not present yet

Comment (macobo) [reply ee/clickhouse/models/test/test_dead_letter_queue.py]: Clickhouse indeed supports (ignores) unknown settings.

**Review by macobo — APPROVED** [review]

**Pull request merged**